### PR TITLE
New package: stw-0.3

### PR DESCRIPTION
**Pull request by wael444**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)

### CODE REVIEW

**Comment by wael444** [comment]
i find it quite confusing building is failing ( missing libXft ) for other architectures despite being `libXft-devel` being specified in `makedepends`

File: srcpkgs/stw/template [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'stw'
+pkgname=stw
+version=0.1
+revision=1
+build_style=gnu-makefile
+make_use_env=yes
+makedepends="libXft-devel freetype-devel base-devel"
```
Comment (paper42) [inline srcpkgs/stw/template:7]: adding pkg-config to hostmakedepends could fix cross
```suggestion
hostmakedepends="pkg-config"
makedepends="libXft-devel freetype-devel"
```

**Comment by sineemore** [comment]
```
[user@lemur void-packages]$ find masterdir/ -name Xft.h
masterdir/usr/aarch64-linux-musl/usr/include/X11/Xft/Xft.h
```

Wrong include path I guess.

**Comment by wael444** [comment]
i have set `make_use_env=yes` in the template to force using the provided `Makefile` in stw so i can avoid errors like these:
```make
In file included from stw.c:14:
/usr/aarch64-linux-gnu/usr/include/X11/Xft/Xft.h:39:10: fatal error: ft2build.h: No such file or directory
   39 | #include <ft2build.h>
      |          ^~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:22: stw.o] Error 1
```
but removing the downside is having the builder add its own flags which also add `-I/usr/aarch64-linux-gnu/usr/include` to the make flags when building, hmm...

**Comment by sineemore** [comment]
In case you feel to patch upstream Makefile we can continue discussion in https://github.com/sineemore/stw/issues/13

Makefile should probably be rewritten I guess.

**Comment by sineemore** [comment]
I think you can try testing the build locally with `./xbps-src pkg -a ...`

**Comment by wael444** [comment]
i know but, this is just a more confusing error.
```
cc -std=c99 -pedantic -Wall -Werror -I/usr/aarch64-linux-gnu/usr/include/freetype2 -I/usr/aarch64-linux-gnu/usr/include/X11 -I/usr/aarch64-linux-gnu/usr/include/X11/Xft -I/usr/aarch64-linux-gnu/usr/include/X11/Xft -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_C_SOURCE=2 -c stw.c
stw.c:14:10: fatal error: X11/Xft/Xft.h: No such file or directory
   14 | #include <X11/Xft/Xft.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:8: stw.o] Error 1
```
`-I/usr/aarch64-linux-gnu/usr/include/X11 -I/usr/aarch64-linux-gnu/usr/include/X11/Xft -I/usr/aarch64-linux-gnu/usr/include/X11/Xft ` is literally in the build flags...?

File: srcpkgs/stw/template [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Template file for 'stw'
+pkgname=stw
+version=0.2
+revision=1
+build_style=gnu-makefile
+make_use_env=compliant
```
Comment (paper42) [inline srcpkgs/stw/template:6]:
```suggestion
make_use_env=yes
```

Comment (wael444) [reply srcpkgs/stw/template:6]: this wont change anything but i guess i'll add it.

Comment (wael444) [reply srcpkgs/stw/template]: > feel free to open a PR to use yes in them (one commit per package, no revbumps).

i have no idea how to do that and git is scary, so..
also yeah thank you for pointing that out.

**Comment by paper42** [comment]
You will need to add `-I/usr/aarch64-linux-gnu/usr/include/` to the command. `-I/usr/aarch64-linux-gnu/usr/include/X11` will try to search for `<X11/Xft/Xft.h>` in `/usr/aarch64-linux-gnu/usr/include/X11/X11/Xft/Xft.h` which obviously doesn't exist

**Comment by wael444** [comment]
> You will need to add -I/usr/aarch64-linux-gnu/usr/include/ to the command. -I/usr/aarch64-linux-gnu/usr/include/X11 will try to search for <X11/Xft/Xft.h> in /usr/aarch64-linux-gnu/usr/include/X11/X11/Xft/Xft.h which obviously doesn't exist

```
cc -std=c99 -pedantic -Wall -Werror -I/usr/include/X11 `pkg-config --cflags fontconfig` -I/usr/aarch64-linux-gnu/usr/include -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_C_SOURCE=2 -c stw.c
In file included from stw.c:3:
/usr/aarch64-linux-gnu/usr/include/limits.h:124:3: error: #include_next is a GCC extension [-Werror]
  124 | # include_next <limits.h>
      |   ^~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:8: stw.o] Error 1
```
note: `-Wall` in `CFLAGS`

**Comment by sineemore** [comment]
@paper42 @wael444 thanks for investigation and packaging ^.^

**Pull request closed without merging**